pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -524,11 +524,11 @@ func recoverHead(dir string, heads *tenantHeads, wals []WALIdentifier) error {
 				if len(rec.Chks.Chks) > 0 {
 					tenant, ok := seriesMap[rec.UserID]
 					if !ok {
-						return errors.New("found tsdb chunk metas without user in WAL replay")
+						return fmt.Errorf("found tsdb chunk metas without user in WAL replay (period=%s): %+v", id.String(), *rec)
```

Comment:
Just double checking `%+v` of this complex nested struct is what we need in error message?

```
type WALRecord struct {
	UserID      string
	Series      record.RefSeries
	Fingerprint uint64
	Chks        ChunkMetasRecord
}

```

Reply:
Yeah, see https://go.dev/play/p/a-aLqq7s_Ix